test/active_record_doctor/detectors/incorrect_length_validation_test.rb

```diff
@@ -61,6 +61,40 @@ def test_no_validation_and_no_limit_is_ok
     refute_problems
   end
 
+  def test_array_inclusion_validation_with_shorter_values_and_limit_is_ok
+    Context.create_table(:users) do |t|
+      t.string :status, limit: 64
+    end.define_model do
+      validates :status, inclusion: { in: ["new", "vip", "a" * 64] }
+    end
+
+    refute_problems
+  end
+
+  def test_array_inclusion_validation_with_longer_values_and_limit_is_error
+    Context.create_table(:users) do |t|
+      t.string :status, limit: 64
+    end.define_model do
+      validates :status, inclusion: { in: ["new", "vip", "a" * 65] }
+    end
+
+    assert_problems(<<~OUTPUT)
+      the schema limits users.status to 64 characters but there's no length validator on Context::User.status - remove the database limit or add the validator
+    OUTPUT
+  end
+
+  def test_proc_inclusion_validation_and_limit_is_error
+    Context.create_table(:users) do |t|
+      t.string :status, limit: 64
+    end.define_model do
+      validates :status, inclusion: { in: ->(user) { STATUSES[user.country] } }
```

Comment:
If I understand correctly we only care about the fact that `in` has received a `Proc`. If that's the case I suggest we just pass a proc with an empty body. I was a bit confused by that code in there.